app/controllers/concerns/hal_actions/cache.rb

```diff
@@ -0,0 +1,31 @@
+module HalActions::Cache
+
+  private
+
+  def index_cache_path
+    PagedCollection.new(
+      filtered(paged(sorted(scoped(resources_base)))),
+      request,
+      item_class: self.class.resource_class,
+      item_decorator: self.class.resource_representer
+    ).cache_key
+  end
+
+  def show_cache_path
+    show_resource.updated_at.utc.to_i
+  end
+
+  module ClassMethods
+    def cache_api_action(action, options = {})
+      options = cache_options.merge(options || {})
+      cache_path_method = options.delete(:cache_path_method) || "#{action}_cache_path"
+      options[:cache_path] = ->(c){ c.valid_params_for_action(action).merge({_c: self.send(cache_path_method) }) } if !options[:cache_path]
```

Comment:
Line is too long. [139/80]<br>Redundant curly braces around a hash parameter.<br>Redundant `self` detected.<br>Space missing to the left of {.<br>Space inside { missing.